File: app/src/main/java/de/bornholdtlee/defaultproject/injection/modules/DatabaseModule.java
package de.bornholdtlee.defaultproject.injection.modules;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import de.bornholdtlee.defaultproject.DefaultApplication;
import de.bornholdtlee.defaultproject.model.MyObjectBox;
import io.objectbox.BoxStore;

@Module
public class DatabaseModule {

    @Provides
    @Singleton
    public BoxStore provideDatabase(DefaultApplication application) {
        return MyObjectBox.builder().androidContext(application).build();
    }
//
//    @Provides
//    @Singleton
//    public Box<Schema> provideSchemaBox(BoxStore store) {
//        return store.boxFor(Schema.class);
//    }
//
//    @Provides
//    @Singleton
//    public DeviceTypeDBHandler provideDeviceScanTypeDBHandler(DBSystelApplication application) {
//        return new DeviceTypeDBHandler(application);
//    }

}

File: app/src/main/java/de/bornholdtlee/defaultproject/injection/modules/DatabaseModule.kt
package de.bornholdtlee.defaultproject.injection.modules

import dagger.Module
import dagger.Provides
import de.bornholdtlee.defaultproject.base.BaseApplication
import de.bornholdtlee.defaultproject.model.DefaultModel
import de.bornholdtlee.defaultproject.model.MyObjectBox
import io.objectbox.Box
import io.objectbox.BoxStore
import javax.inject.Singleton

@Module
class DatabaseModule {

    @Provides
    @Singleton
    fun provideDatabase(application: BaseApplication): BoxStore {
        //        BoxStore boxStore = MyObjectBox.builder().androidContext(application).build();
        //        if (BuildConfig.DEBUG) {
        //            new AndroidObjectBrowser(boxStore).constructionsStart(application);
        //        }
        return MyObjectBox.builder().androidContext(application).build()
    }

    @Provides
    @Singleton
    fun provideDefaultBox(store: BoxStore): Box<DefaultModel> {
        return store.boxFor(DefaultModel::class.java)
    }

}
